src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/TypedContentTypeBuilder.cs

```diff
@@ -57,13 +58,14 @@ public void Build(FieldType contentQuery, ContentTypeDefinition contentTypeDefin
                 }
 
                 var partType = _partTypes.GetOrAdd(part.PartDefinition.Name, key => typeActivator.GetTypeActivator(key).Type);
-                var queryGraphType = _partObjectGraphTypes.GetOrAdd(part.PartDefinition.Name,
-                                                                    partName =>
-                                                                    {
-                                                                        queryObjectGraphTypes ??= serviceProvider.GetService<IEnumerable<IObjectGraphType>>();
-                                                                        return queryObjectGraphTypes?.FirstOrDefault(x => x.GetType().BaseType.GetGenericArguments().First().Name == partName);
-                                                                    }
-                                                                );
+                var queryGraphType = _partObjectGraphTypes
+                    .GetOrAdd(part.PartDefinition.Name,
+                        partName =>
+                        {
+                            queryObjectGraphTypes ??= serviceProvider.GetServices<IObjectGraphType>();
```

Comment:
The problem might be here. `seviceProvider` is stored in the closure. It should be resolved within this method and not outside.

Reply:
I think we should not store the result of the queryObject outside that method. Yes. But that did not solve the issue.

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/726a7432-4f15-48b2-a96f-103f22835efd)